### PR TITLE
Fixes #368 Preserve null and empty arrays in unwind

### LINKED ISSUE #368: Application crashed when there are no logs for a repository
### This is a
- [x] Bug report
- [ ] Feature request

### Description

When we register the application to Yaydoc and then access its logs without creating any, the application crashes. This is happening since we use MongoDB lookup. Unwinding the logs inside the repository, an empty result is returned when the repository is empty. 

The expected behaviour would be to show a page similar to how travis shows.
image

### Steps to reproduce
1. Register a repository to Yaydoc
2. Open `/owner/reponame` or `/owner/reponame/logs`

### PR DESCRIPTION
**Pull request by imujjwal96**
## Description

* Update Repository model, using `preserveNullAndEmptyArray` key with unwind
* Add templates for `Repository Not Found` and `No Build Logs`
* Update `title` for routes
## Related Issue

issue #368 

## How Has This Been Tested?

https://yaydoc6.herokuapp.com
## Screenshots (if appropriate):
image
image

## Types of changes

- [x] Bug fix 
- [ ] New feature

## Checklist:

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only one commit per issue.

### CODE REVIEW

**Comment by imujjwal96** [comment]
@pri22296 @sch00lb0y Your reviews?

File: views/repository/204.jade [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    .error-code.m-b-10.m-t-20
+      | 204
+      i.fa.fa-warning
+    h3.font-bold
+      | We could not find any build logs for&nbsp;
+      span(style="color: #db4545")=name
+    .error-desc
+      | It is possible that the repository is registered but there have been no commits afterwards!
+      div
+        a.login-detail-panel-button.btn(href='/')
```
Comment (pri22296) [inline views/repository/204.jade]: Would be better to send back to dashboard instead of homepage.

File: views/repository/404.jade [hunk 2]
```diff
@@ -1 +1,17 @@
...
+      i.fa.fa-warning
+    h3.font-bold
+      | We couldn't find the repository&nbsp;
+      span(style="color: #db4545")=name
+    .error-desc
+      | Either the repository does not exist or it is not registered to Yaydoc!
+      div
+        a.login-detail-panel-button.btn(href='/')
+          i.fa.fa-arrow-left
+          |                         Go back to Homepage
```
Comment (pri22296) [inline views/repository/404.jade]: here too

File: views/repository/204.jade (hunk 1 above)
Comment (imujjwal96) [inline views/repository/204.jade]: anybody can view the logs for any repository. If a user is not logged in, there is no point of redirecting to dashboard

Comment (pri22296) [inline views/repository/204.jade]: Good point about non-logged in users. Would it be possible to send logged in users to dashboard and non-logged in users to preview page? It's just that it was slightly irritating to having to click `dashboard`, then `settings` everytime.

Comment (sch00lb0y) [inline views/repository/204.jade]: I think if user not logged it'll automatically redirect to `\`

**Comment by imujjwal96** [comment]
@pri22296 @sch00lb0y Updated.

**Pull request merged**